### DISCUSSION TITLE
Support for 'Train on inputs' that handles interactive discussions

### DISCUSSION THREAD

**Discussion by araleza** [axolotl-ai-cloud/axolotl#1011] [Q&A] [answered]
Hi,

I'm aware of the 'Train on inputs' boolean that allows the training to only generate loss for the replies to instructions, rather than on the instructions themselves.  But, how could this be made to work for extended conversations between a user and the AI?  Especially if I have a raw input where the conversation looks like:

User: Here is my question
AI: Here is the reply
User: Here is another question, based on your first reply
AI: Here is a more detailed reply in response

I want the training to learn both of the AI lines without learning either of the User's lines.  Obviously I could break this into two pairs, but I wouldn't actually want the second AI reply to be learned as a response to the second User question without the first question and response being part of the context.

**Comment by NanoCode012** [comment, marked as answer]
The default behavior is that it'll mask both user lines as you wanted. It doesn't mask only last and does not require you to split into two.
